mslib/msui/mscolab.py

```diff
@@ -219,6 +219,7 @@ def connect_handler(self):
             s.auth = auth
             s.headers.update({'x-test': 'true'})
             r = s.get(url_join(url, 'status'), timeout=(2, 10))
+            idp_enabled  = json.loads(r.text)["IDP_ENABLED"]
```

Comment:
```suggestion
            try:
                idp_enabled = json.loads(r.text)["IDP_ENABLED"]
            except (json.decoder.JSONDecodeError, KeyError):
                idp_enabled = false
```

The current version would crash if we were using this against an older version of mscolab, since it does not return well-formed json and/or the IDP_ENABLED key would not be present. This should also be moved further down, into the branch checking `r.status_code == 200`, since we can only expect the json response in the case of a successful request.